src/newProject/NewProjectPanel.ts

```diff
@@ -0,0 +1,335 @@
+// Copyright 2019 Espressif Systems (Shanghai) CO LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ensureDir, copy } from "fs-extra";
+import { basename, delimiter, join } from "path";
+import * as vscode from "vscode";
+import { createNewProjectHtml } from "./createNewProjectHtml";
+import * as idfConf from "../idfConfiguration";
+import { LocDictionary } from "../localizationDictionary";
+import { INewProjectArgs } from "./newProjectInit";
+import { getExamplesList, IExample } from "../examples/Examples";
+import {
+  IMetadataFile,
+  getToolsInMetadataForIdfPath,
+  IPath,
+  ITool,
+} from "../ITool";
+import * as utils from "../utils";
+import { Logger } from "../logger/logger";
+import { OutputChannel } from "../logger/outputChannel";
+import { IComponent } from "../espIdf/idfComponent/IdfComponent";
+
+const locDictionary = new LocDictionary("NewProjectPanel");
+
+export class NewProjectPanel {
+  public static currentPanel: NewProjectPanel | undefined;
+
+  public static createOrShow(
+    extensionPath: string,
+    newProjectArgs?: INewProjectArgs
+  ) {
+    const column = vscode.window.activeTextEditor
+      ? vscode.window.activeTextEditor.viewColumn
+      : vscode.ViewColumn.One;
+    if (NewProjectPanel.currentPanel) {
+      NewProjectPanel.currentPanel.panel.reveal(column);
+    } else {
+      NewProjectPanel.currentPanel = new NewProjectPanel(
+        extensionPath,
+        newProjectArgs,
+        column
+      );
+    }
+  }
```

Comment:
For me the `Create new Project` is not rendering any UI

<img width="1552" alt="Screen Shot 2020-04-27 at 1 12 34 PM" src="https://user-images.githubusercontent.com/25585652/80347299-c1b54180-8889-11ea-97ba-359889461e04.png">

Reply:
Any errors in Developer Tools/ Logs/ ESP-IDF channel?

Reply:
None actually ... which is strange...  I will debug it and let you know soon.

Reply:
I found the root cause of this, is that when `IDF_PATH` is set incorrectly it just crash silently and not render

Possible point of failure:
`public async loadInitialMetadata(metadata: IMetadataFile) {...}`

Reply:
Also crashing here 
```js
const pyVenvList = metadata.venv.filter(...
```